src/base.ts

```diff
@@ -100,7 +100,10 @@ export abstract class Step implements BaseStep, Serializable {
   }
 
   async toJson(
-    opts: ToJsonSerializationOptions = { explicitDependencies: false },
+    opts: ToJsonSerializationOptions = {
+      explicitDependencies: false,
+      acceptAllConditions: false,
+    },
```

Comment:
we should probably define

```ts
const defaultOptions: ToJsonSerializationOptions = {
  explicitDependencies: false,
  acceptAllConditions: false,
};
```
and then 
```suggestion
    opts: ToJsonSerializationOptions = { ...defaultOptions },
```

Reply:
That sounds good!